### PR TITLE
Add a core generated REST API document

### PR DESCRIPTION
**Pull request by smarterclayton**

### CODE REVIEW

**Comment by smarterclayton** [comment]
@adellape these documents have crazy TOCs - what are my options for simplifying them (basically they have hundreds of sub items)?

**Comment by smarterclayton** [comment]
For instance, I think I only want the first level titles.

**Comment by adellape** [comment]
@smarterclayton Looks like `toclevels` works (http://asciidoctor.org/docs/user-manual/#levels):

```
= OpenShift v1 REST API
{product-author}
{product-version}
:data-uri:
:icons:
:experimental:
:toc: macro
:toc-title:
:toclevels: 1

toc::[]
```

File: rest_api/overview.adoc [hunk 1]
```diff
@@ -4,8 +4,11 @@
...
+:toc: macro
+:toc-title:
 
-If you'd like to contribute to OpenShift documentation, see our
-https://github.com/openshift/openshift-docs[source repository] and
-https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc[guidelines]
-to get started.
+The OpenShift distribution of Kubernetes includes the link:./kubernetes_v1/overview.html[Kubernetes v1 REST API]
+and the link:./openshift_v1/overview.html[OpenShift v1 REST API]. These are RESTful APIs accessible via HTTP(s)
+on the OpenShift master servers.
```
Comment (adellape) [inline rest_api/overview.adoc]: Did you want these to link to the other two new topics? If so they can be `link:./kubernetes_v1.html` and `link:./openshift_v1.html`, respectively.

Comment (smarterclayton) [inline rest_api/overview.adoc]: Oops, yeah.  Moved stuff.

----- Original Message -----

**Comment by smarterclayton** [comment]
Both fixed, PTAL

----- Original Message -----

**Comment by adellape** [comment]
LGTM, and the new `import_api` task worked as advertised (in https://github.com/openshift/origin/blob/master/api/README.md).

@nhr Can you ack the Rakefile additions?

**Comment by nhr** [comment]
Looks good. @smarterclayton thank you for putting this together.

**Comment by adellape** [comment]
@smarterclayton Fine to merge after build passes?

**Comment by smarterclayton** [comment]
Yes

**Comment by smarterclayton** [comment]
Looks like the build passed once, but then didn't get reported.

**Comment by CowboysFan** [comment]
@smarterclayton @nhr That's a long document without having a TOC and not having the ability to navigate through it. What does it look like when it's generated? Can we still restructure it, and will that cause any issues with the automatic generation?

**Comment by smarterclayton** [comment]
> On Jun 3, 2015, at 3:10 AM, Bilhar Aulakh <notifications@github.com> wrote:
> 
> @smarterclayton @nhr That's a long document without having a TOC and not having the ability to navigate through it. What does it look like when it's generated? 
> 
There are several hundred topics.
> Can we still restructure it, and will that cause any issues with the automatic generation?
> 
At this point we are going to be iterating on flushing out documentation, and it's more valuable to be able to generate the doc automatically than to have it be broken up.  Maybe in the future we can add some transformation / splits, although it'll be past 3.0z

**Pull request merged**